src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -1406,6 +1406,32 @@ public void CopyFileOnItself2(bool isUseHardLinks, bool isUseSymbolicLinks)
             }
         }
 
+        [Theory]
+        [MemberData(nameof(GetHardLinksSymLinks))]
+        public void CopyFileItselfNonExistentFile(bool isUseHardLinks, bool isUseSymbolicLinks)
```

Comment:
This behavior doesn't care if you're using hard links or symlinks or neither, right? Why do you need to run it four times?

Also:
```suggestion
        public void CopyNonexistentFileOntoItselfThrows(bool isUseHardLinks, bool isUseSymbolicLinks)
```